test/fork/smoke-test/Smoke_Deleverage.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: BUSL-1.1
+// This code is made available under the terms and conditions of the Business Source License 1.1 (BUSL-1.1).
+// The act of publishing this code is driven by the aim to promote transparency and facilitate its utilization for educational purposes.
+
+pragma solidity 0.8.18;
+
+import { Smoke_Base } from "./Smoke_Base.t.sol";
+import { IConfigStorage } from "@hmx/storages/interfaces/IConfigStorage.sol";
+import { IPerpStorage } from "@hmx/storages/interfaces/IPerpStorage.sol";
+import { IEcoPythCalldataBuilder } from "@hmx/oracles/interfaces/IEcoPythCalldataBuilder.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { PythStructs } from "pyth-sdk-solidity/IPyth.sol";
+import { HMXLib } from "@hmx/libraries/HMXLib.sol";
+
+import "forge-std/console.sol";
+
+contract Smoke_Liquidate is Smoke_Base {
+  uint256 internal constant BPS = 10_000;
+
+  address internal constant USDC = 0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8;
+  address internal constant TRADE_SERVICE = 0xcf533D0eEFB072D1BB68e201EAFc5368764daA0E;
+
+  function setUp() public virtual override {
+    super.setUp();
+
+    vm.prank(OWNER);
+    configStorage.setLiquidityConfig(
+      IConfigStorage.LiquidityConfig({
+        depositFeeRateBPS: 30, // 0.3%
+        withdrawFeeRateBPS: 30, // 0.3%
+        maxHLPUtilizationBPS: 8000, // 80%
+        hlpTotalTokenWeight: 0,
+        hlpSafetyBufferBPS: 10000, // 100%
+        taxFeeRateBPS: 50, // 0.5%
+        flashLoanFeeRateBPS: 0,
+        dynamicFeeEnabled: true,
+        enabled: true
+      })
+    );
+  }
+
+  function testCorrectness_SmokeTest_deleverageFirstTen() external {
+    IEcoPythCalldataBuilder.BuildData[] memory data = _buildDataForPrice_Deleverage();
+    (
+      uint256 _minPublishTime,
+      bytes32[] memory _priceUpdateCalldata,
+      bytes32[] memory _publishTimeUpdateCalldata
+    ) = ecoPythBuilder.build(data);
+
+    IPerpStorage.Position[] memory positions = perpStorage.getActivePositions(10, 0);
+
+    if (positions.length == 0) {
+      console.log("No position to be deleveraged");
+      return;
+    }
```

Comment:
We want to test deleverage. If it has nothing to deleverage, please revert so we know something wrong.